### ISSUE TITLE
k3d create command fails at installing Argo

### ISSUE THREAD

**Issue by tomekz** [kubefirst/kubefirst#1839]
### Which version of kubefirst are you using?

04b3cef515ca5602e0ed4bf06b6e05ebdfd2641e

### Which cloud provider?

k3d (local)

### Which DNS?

None specific

### Which installation type?

CLI

### Which distributed Git provider?

GitHub

### What is the issue?

The cluster provisioning fails with the following output:

04b3cef515ca5602e0ed4bf06b6e05ebdfd2641e

```
Running preflight checks                ... done! [5 in 48ms]
Cloning and formatting git repositories ... done! [1 in 0s]
Applying github Terraform               ... done! [1 in 0s]
Pushing git repositories                ... done! [1 in 0s]
Creating k3d cluster                    ... done! [1 in 0s]
Bootstrapping Kubernetes resources      ... done! [2 in 47ms]
Verifying Kubernetes cluster is ready   ... done! [3 in 24.308s]
Installing and configuring Argo CD      ...  0.0% [......................................] [0 in 37.293296s]
[#############################################################################...........] [1m2s]
Installing and configuring Argo CD      ...  0.0% [......................................] [0 in 1m9.293236s]
[#############################################################################...........] [1m34s]
E1014 22:01:44.053542   63059 memcache.go:104] couldn't get resource list for metrics.k8s.io/v1beta1: Get "https://
Installing and configuring Argo CD      ...  0.0% [......................................] [0 in 1m23.293176s]
[#############################################################################...........] [1m48s]
Error: error applying Namespace argocd: Patch "https://0.0.0.0:62506/api/v1/namespaces/argocd?fieldManager=kubefirst": http2: client connection lost

If a detailed error message was available, please make the necessary corrections before retrying.
You can re-run the last command to try the operation again.

```

logs:
```
2023-10-14T22:00 INF pkg/helpers.go:78 > Using config file: /Users/tomasz.zadrozny/.kubefirst
2023-10-14T22:00 INF cmd/k3d/create.go:131 > github.com ssh-ed25519

2023-10-14T22:00 INF cmd/k3d/create.go:215 > verifying github authentication
2023-10-14T22:00 INF kubefirst-api/pkg/handlers/GitHubHandler/github.go:173 > GitHub user: tomekz
2023-10-14T22:00 INF cmd/k3d/create.go:383 > kubefirst version configs.K1Version: development
2023-10-14T22:00 INF cmd/k3d/create.go:384 > cloning gitops-template repo url: https://github.com/kubefirst/gitops-template.git
2023-10-14T22:00 INF cmd/k3d/create.go:385 > cloning gitops-template repo branch: main
2023-10-14T22:00 INF cmd/k3d/create.go:394 > checking authentication to required providers
2023-10-14T22:00 INF cmd/k3d/create.go:447 > already completed github checks - continuing
2023-10-14T22:00 INF cmd/k3d/create.go:473 > already setup kbot user - continuing
2023-10-14T22:00 INF cmd/k3d/create.go:477 > validation and kubefirst cli environment check is complete
2023-10-14T22:00 INF cmd/k3d/create.go:548 > already completed download of dependencies to `$HOME/.k1/tools` - continuing
2023-10-14T22:00 INF cmd/k3d/create.go:595 > already completed gitops repo generation - continuing
2023-10-14T22:00 INF cmd/k3d/create.go:639 > already created GitHub Terraform resources
2023-10-14T22:00 INF cmd/k3d/create.go:687 > referencing gitops repository: git@github.com:tomekz/gitops.git
2023-10-14T22:00 INF cmd/k3d/create.go:688 > referencing metaphor repository: https://github.com/tomekz/metaphor.git
2023-10-14T22:00 INF cmd/k3d/create.go:752 > already pushed detokenized gitops repository content
2023-10-14T22:00 INF cmd/k3d/create.go:781 > already created k3d cluster resources
2023-10-14T22:00 INF cmd/k3d/create.go:817 > already added secrets to k3d cluster
2023-10-14T22:00 ERR internal/gitShim/containerRegistryAuth.go:61 > error while creating secret for container registry auth: secrets "container-registry-auth" already exists
2023-10-14T22:00 INF pkg/k8s/exec.go:179 > waiting for traefik Deployment to be created
2023-10-14T22:00 INF pkg/k8s/exec.go:327 > waiting for traefik Deployment to be ready - this could take up to 240 seconds
2023-10-14T22:00 INF pkg/k8s/exec.go:342 > all Pods in Deployment traefik are ready
2023-10-14T22:00 INF pkg/k8s/exec.go:179 > waiting for metrics-server Deployment to be created
2023-10-14T22:00 INF pkg/k8s/exec.go:327 > waiting for metrics-server Deployment to be ready - this could take up to 240 seconds
2023-10-14T22:00 INF pkg/k8s/exec.go:342 > all Pods in Deployment metrics-server are ready
2023-10-14T22:00 INF cmd/k3d/create.go:911 > installing argocd
2023-10-14T22:00 INF runtime/pkg/k8s/KubernetesClient/apply.go:37 > applying objects against kubernetes cluster

```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jarededwards** [comment]
hi @tomekz ! thanks for reporting this, it looks like you're currently running off of the main branch - can you confirm you are not running off of the latest published binary `v2.2.17`?
```
2023-10-14T22:00 INF cmd/k3d/create.go:383 > kubefirst version configs.K1Version: development
2023-10-14T22:00 INF cmd/k3d/create.go:384 > cloning gitops-template repo url: https://github.com/kubefirst/gitops-template.git
2023-10-14T22:00 INF cmd/k3d/create.go:385 > cloning gitops-template repo branch: main
```
our main branch is not always stable and we're just a few days away from a new release that introduces breaking changes to repository structures to accommodate our cluster management feature set

**Comment by tomekz** [comment]
Yes I'm using the main branch. I will switch to latest release in that case. Thnx for the help
